app/services/CheckoutService.scala

```diff
@@ -61,6 +61,22 @@ class CheckoutService(identityService: IdentityService,
     case _ => false
   }
 
+  def processIntroductoryPeriod(defaultPaymentDelayt: Days, originalCommand: Subscribe): Subscribe = {
+    val additionalRateplan = (originalCommand.ratePlans.head.productRatePlanId match {
+      case catalog.weeklyZoneA.sixWeeks.id.get => Some(catalog.weeklyZoneA.quarter)
+      case catalog.weeklyZoneC.sixWeeks.id.get => Some(catalog.weeklyZoneC.quarter)
+      case _ => None
+    }).map(plan => RatePlan(plan.id.get, None))
```

Comment:
not here, but I don't like that we use `get` rather than `value` as it makes it more likely we'll miss a `None.get` in the code